### PR TITLE
Batch blocking

### PR DESCRIPTION
**Pull request by EmileSonneveld**
- Moved the validation from sync job to load_collection (this will check more when running as a sync job)
- Made the dry-run in /validate more complete so the same validation code as in load_collection can be used
- Fixed a case where dates with and without timezones got mixed
- Removed need for verify_for_synchronous_processing
- Allow to not specify spatial_extent. Fallback to extent from layercatalog or world extent
- Allow to not specify temporal_extent. Fallback to extent from layercatalog or 1970-2070

Some tests fail, because it needs this MR too: https://github.com/Open-EO/openeo-python-driver/pull/295

### CODE REVIEW

**Comment by EmileSonneveld** [comment]
@soxofaan would it make sense to keep `do_extent_check` on False by default, so we can deploy more early. And change it to true after vacation?

**Review by soxofaan — COMMENTED** [review]
some quick notes

File: openeogeotrellis/backend.py [hunk 1]
```diff
@@ -1203,135 +1201,16 @@ class GpsProcessing(ConcreteProcessing):
...
-        catalog = env.backend_implementation.catalog
-        allow_check_missing_products = smart_bool(env.get("allow_check_missing_products", True))
-        sync_job = smart_bool(env.get("sync_job", False))
-        large_layer_threshold_in_pixels = int(float(env.get("large_layer_threshold_in_pixels", LARGE_LAYER_THRESHOLD_IN_PIXELS)))
-
-        for source_id, constraints in source_constraints:
-            source_id_proc, source_id_args = source_id
-            if source_id_proc == "load_collection":
+        try:
+            for source_constraint in source_constraints.copy():  # copy because _extract_load_parameters is stateful
```
[the remark this answers is not in the document]
Comment (EmileSonneveld) [reply openeogeotrellis/backend.py]: Changed to deepcopy

File: openeogeotrellis/layercatalog.py [hunk 2]
```diff
@@ -147,6 +152,24 @@ def create_datacube_parameters(self, load_params, env):
...
+            # Only care for certain errors and make list of strings:
+            issues = [e["message"] for e in issues if e["code"] == "ExtentTooLarge"]
+            if issues:
+                if env.get("sync_job", False):
+                    raise ProcessGraphComplexityException(
+                        ProcessGraphComplexityException.message + f" Reasons: {' '.join(issues)}"
+                    )
+                else:
+                    raise ProcessGraphComplexityException(
+                        "Found errors in process graph. Disable this check with 'job_options.do_extent_check': " +
```
[the remark this answers is not in the document]
Comment (EmileSonneveld) [reply openeogeotrellis/layercatalog.py]: Changed to `Process graph contains nodes with a too large spatiotemporal extent and will likely time out. Disable this check with...`

**Comment by EmileSonneveld** [comment]
Tested by running the python tests locally, and the collection of process graphs from CDSE and Terrascope.

File: openeogeotrellis/layercatalog.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (EmileSonneveld) [reply openeogeotrellis/layercatalog.py]: ProcessGraphComplexityException is based on: https://github.com/Open-EO/openeo-api/blob/master/errors.json#L244

File: openeogeotrellis/geopysparkcubemetadata.py [hunk 3]
```diff
@@ -118,6 +118,32 @@ def get_nodata_value(self, requested_bands, default_value) -> float:
             no_data_value = default_value
         return float(no_data_value)
 
+    def get_layer_crs(self):
+        crs = self.get("cube:dimensions", "x", "reference_system", default="EPSG:4326")
+        if isinstance(crs, int):
+            crs = "EPSG:%s" % str(crs)
+        elif isinstance(crs, dict):
+            if crs["name"] == "AUTO 42001 (Universal Transverse Mercator)":
```
Comment (JeroenVerstraelen) [inline openeogeotrellis/geopysparkcubemetadata.py:126]: This could also use a dict.get to avoid throwing an exception.

**Review by JeroenVerstraelen — APPROVED** [review]

File: openeogeotrellis/geopysparkcubemetadata.py (hunk 3 above)
Comment (EmileSonneveld) [reply openeogeotrellis/geopysparkcubemetadata.py:126]: True! Changed here: https://github.com/Open-EO/openeo-geopyspark-driver/commit/cdd731ce6d684eba894beff7c8ac78266ddf12b0

**Pull request merged**